add release note of e5.5.0 (#2338)
emqtt/emq-docs-cn: en_US/changes/changes-ee-v5.md

# Version 5

## 5.5.0

> *Release Date: 2024-01-31*

Reviewer: ```suggestion
*Release Date: 2024-02-01*
```

### Enhancements

> - [#12085](https://github.com/emqx/emqx/pull/12085) EMQX has been upgraded to leverage the capabilities of OTP version 26.1.2-2. 

Reviewer 2: ```suggestion
- [#12085](https://github.com/emqx/emqx/pull/12085) EMQX has been upgraded to leverage the capabilities of OTP version 26.1.2-2. NOTE: Docker images are still built with OTP 25.3.2.
```

  - Added `/api/v5/prometheus/auth` endpoint to provide metrics such as execution count and running status for all authenticatiors and authorizators.
  - Added `/api/v5/prometheus/data_integration` endpoint to provide metrics such as execution count and status for all rules, actions, and connectors.

  Limitations:
  Prometheus push gateway only supports the content in `/api/v5/prometheus/stats?mode=node`.

  For more API details and metric type information, please see swagger api docs.

- [#12196](https://github.com/emqx/emqx/pull/12196) Improved network efficiency during routes cleanup.

  Previously, when a node was down, a delete operation for each route to that node must be exchanged between all the other live nodes. After this change, only one `match and delete` operation is exchanged between all live nodes, significantly reducing the number of necessary network packets and decreasing the load on the inter-cluster network.
  This optimization must be especially helpful for geo-distributed EMQX deployments where network latency can be significantly high.
  

> - [#12354](https://github.com/emqx/emqx/pull/12354) Applied post-config bridge changes in parallel. This can greatly improve the performance when multiple bridges are being changed, for example, when a backup file is being imported.

Author: ```suggestion
- [#12354](https://github.com/emqx/emqx/pull/12354) Applied post-config bridge changes in parallel. This can greatly improve the performance when multiple bridges are being changed, for example, when a backup file is being imported.

- [#12396](https://github.com/emqx/emqx/pull/12396) Enhanced the `authentication/:id/import_users` interface for a more user-friendly user import feature:

  - Add new parameter `?type=plain` to support importing users with plaintext passwords in addition to the current solution which only supports password hash.
  - Support `content-type: application/json` to accept HTTP Body in JSON format in extension to the current solution which only supports `multipart/form-data` for csv format.
```

- [#12269](https://github.com/emqx/emqx/pull/12269) Improved error handling in the `/clients` interface; now returns a 400 status with more detailed error messages, instead of a generic 500, for query string validation failures.

- [#12285](https://github.com/emqx/emqx/pull/12285) Updated the CoAP gateway to support short parameter names for slight savings in datagram size. For example, `clientid=bar` can be written as `c=bar`.
  
- [#12303](https://github.com/emqx/emqx/pull/12303) Fixed the message indexing in retainer. Previously, clients with wildcard subscriptions might receive irrelevant retained messages not matching their subscription topics.

- [#12305](https://github.com/emqx/emqx/pull/12305) Corrected an issue with incomplete client/connection information being passed into `emqx_cm`, which could lead to internal inconsistencies and affect memory usage and operations like node evacuation.

- [#12306](https://github.com/emqx/emqx/pull/12306) Fixed an issue preventing the connectivity test for the Connector from functioning correctly after updating the password via the HTTP API.

> - [#12359](https://github.com/emqx/emqx/pull/12359) Fixed an issue causing error messages when restarting a node configured with some types of data bridges.  Additionally, these bridges were at risk of entering a failed state upon node restart, requiring a manual restart to restore functionality.

Author: ```suggestion
- [#12359](https://github.com/emqx/emqx/pull/12359) Fixed an issue causing error messages when restarting a node configured with some types of data bridges.  Additionally, these bridges were at risk of entering a failed state upon node restart, requiring a manual restart to restore functionality.

- [#12404](https://github.com/emqx/emqx/pull/12404) Fixed an issue where restarting a busy data integration could lead to data integration metrics to stop being collected.
```

- [#12282](https://github.com/emqx/emqx/pull/12282) Improved the HTTP API error response for MySQL bridge creation failures. It also resolved a problem with removing MySQL bridges containing undefined columns in their SQL.

- [#12291](https://github.com/emqx/emqx/pull/12291) Fixed inconsistencies in EMQX’s handling of configuration updates involving sensitive parameters, which previously led to stray `"******"` strings in cluster configuration files.

> - [#12292](https://github.com/emqx/emqx/pull/12292) Fixed a crash in the `GET /bridges` API when creating a Syskeeper Forwarder connector and action.

Author: ```suggestion
- [#11902](https://github.com/emqx/emqx/pull/11902) Integrated Nari Syskeeper 2000 as a new bridge backend.
```

# Version 5

## 5.5.0

*Release Date: 2024-01-31*

### Enhancements

- [#12085](https://github.com/emqx/emqx/pull/12085) Upgrade to OTP 26.1.2-2

- [#12189](https://github.com/emqx/emqx/pull/12189) Enhanced JWT ACL Claim Format.

  The JWT ACL claim has been upgraded to support a more versatile format.
  It now accepts an array structure, which resembles the file-based ACL rules.

  For example:

  ```json

>   [
>   {
>   "permission": "allow",
>   "action": "pub",
>   "topic": "${username}/#",
>   "qos": [0,1],
>   "retain": true
>   },
>   {
>   "permission": "allow",
>   "action": "sub",
>   "topic": "eq ${username}/#",
>   "qos": [0,1]
>   },
>   {
>   "permission": "deny",
>   "action": "all",
>   "topics": ["#"]
>   }
>   ]

Reviewer 3: ```suggestion
[
  {
    "permission": "allow",
    "action": "pub",
    "topic": "${username}/#",
    "qos": [0, 1],
    "retain": true
  },
  {
    "permission": "allow",
    "action": "sub",
    "topic": "eq ${username}/#",
    "qos": [0, 1]
  },
  {
    "permission": "deny",
    "action": "all",
    "topics": ["#"]
  }
]
```

>   ```json
>   [
>   {
>   "permission": "allow",
>   "action": "pub",
>   "topic": "${username}/#",
>   "qos": [0,1],
>   "retain": true
>   },
>   {
>   "permission": "allow",
>   "action": "sub",
>   "topic": "eq ${username}/#",
>   "qos": [0,1]
>   },
>   {
>   "permission": "deny",
>   "action": "all",
>   "topics": ["#"]
>   }
>   ]
>   ```

Reviewer 3: ```suggestion
  ```json
[
  {
    "permission": "allow",
    "action": "pub",
    "topic": "${username}/#",
    "qos": [0, 1],
    "retain": true
  },
  {
    "permission": "allow",
    "action": "sub",
    "topic": "eq ${username}/#",
    "qos": [0, 1]
  },
  {
    "permission": "deny",
    "action": "all",
    "topics": ["#"]
  }
]
  ```
```

  Previously the default timeout was 5s which would often be caused by HTTP API calls due to emqx taking too long to join cluster.

  Add a new endpoint `/cluster/:node/invite_async` to support an asynchronous way to invite nodes to join the cluster,
  and a new endpoint `cluster/invitation` to inspect the join status.

- [#12272](https://github.com/emqx/emqx/pull/12272) Added a new endpoint `DELETE /retainer/messages` to `retain` API to clean all retained messages

  Also added an optional topic filter in the query string for the endpoint "GET /retainer/messages", e.g. "topic=t/1".

- [#12277](https://github.com/emqx/emqx/pull/12277) Add `mqtt/delayed/messages/:topic` endpoint to remove delayed messages by topic name.

- [#12278](https://github.com/emqx/emqx/pull/12278) In the REST API, the maximum limit of page queries is adjusted to 10,000.

- [#12289](https://github.com/emqx/emqx/pull/12289) Add new config `authorization.cache.excludes` to support ACL cache exclusion.

  When configured with a list of topic-filters, the publish or subscribe permission check results for a matching topic or topic filter will not be cached.

> - [#12299](https://github.com/emqx/emqx/pull/12299) Expose more metrics to improve observability:

Reviewer 2: I think this entry should be removed in favor of the one on line 94

- [#12329](https://github.com/emqx/emqx/pull/12329) Add `broker.routing.batch_sync` configuration that enables separate process pool used to synchronize subscriptions with the global routing table in a batched manner.
  It's especially useful on nodes interconnected with the cluster through links with non-negligible latency, but might help in other scenarios by ensuring that the broker pool has less chance being overloaded.

- [#12333](https://github.com/emqx/emqx/pull/12333) Add 'tags' field for actions and connectors

  Similar to 'description' field (which is a free text annotation), 'tags' can be used to annotate actions and connectors for filtering/grouping.

- [#12072](https://github.com/emqx/emqx/pull/12072) Supports async query mode for GreptimeDB data bridge. It provides better performance.

- [#12194](https://github.com/emqx/emqx/pull/12194) Improve Kafka producer performance.

- [#12247](https://github.com/emqx/emqx/pull/12247) The bridges for InfluxDB have been split so they are available via the connectors and actions APIs. They are still backwards compatible with the old bridge API.

> - [#12299](https://github.com/emqx/emqx/pull/12299) # Expose more metrics to improve observability:

Reviewer 2: this one is listed 2 times, the same entry is on line 63

  - Add `/api/v5/prometheus/data_integration` endpoint to provide metrics such as execution count and status for all rules, actions, and connectors.

  Limitations:
  Prometheus push gateway only supports the content in `/api/v5/prometheus/stats?mode=node`

  For more API details and metric type information. Please see also in swagger api docs.

- [#12196](https://github.com/emqx/emqx/pull/12196) Improve network efficiency during routes cleanup.

  Previously, when a node node was down, a delete operation for every route to that node must have been exchanged between all other live nodes.
  After this change, only one 'match and delete' operation is exchanged between all live nodes, meaning that much fewer packets are to be sent over inter-cluster network.
  This optimization must be especially helpful for geo-distributed EMQX deployments, when network latency can be significantly high.

- [#12354](https://github.com/emqx/emqx/pull/12354) Apply post config bridge changes in parallel.
  This can greatly improve the performance when multiple bridges are being changed,
  e.g. when a backup file is being imported.

### Bug Fixes

> - [#12232](https://github.com/emqx/emqx/pull/12232) Fixed issue where commit log info was incorrectly retained after forcefully leaving a cluster, removing unnecessary commit log data improves leave operation cleanup.

Reviewer 2: ```suggestion
- [#12232](https://github.com/emqx/emqx/pull/12232) Fixed an issue when cluster commit log table was not deleted after a node was forced to leave a cluster.
```

- [#12305](https://github.com/emqx/emqx/pull/12305) Eliminated passing incomplete client/connection information into `emqx_cm`. This could lead to internal inconsistency and affect memory consumption and some processes like node evacuation.

- [#12306](https://github.com/emqx/emqx/pull/12306) Fixed an issue that prevented probing a bridge from working after changing the password in the HTTP API request.

- [#12359](https://github.com/emqx/emqx/pull/12359) Fixed an issue that could lead to error messages when restarting a node configured with some types of data bridges.  Said bridges could also start in a failed state, requiring manual restart.

- [#12282](https://github.com/emqx/emqx/pull/12282) Improved HTTP API error message when the creation of a MySQL bridge fails.

  Fixed an issue that prevented removing a MySQL bridge when its SQL contained undefined columns.

- [#12291](https://github.com/emqx/emqx/pull/12291) Fix inconsistency in how EMQX handles configuration updates where sensitive parameters are involved, which led to occurrences of stray `"******"` strings in the cluster configuration file.

> - [#12292](https://github.com/emqx/emqx/pull/12292) Fixed `GET /bridges` API crash when a Syskeeper Forwarder bridge exists.

Reviewer 4: This is a newly added data bridge, so it shouldn't be necessary？

> - [#12292](https://github.com/emqx/emqx/pull/12292) Fixed a crash when listing bridges via `GET /bridges` when a Syskeeper Forwarder bridge exists.

Reviewer 2: ```suggestion
- [#12292](https://github.com/emqx/emqx/pull/12292) Fixed `GET /bridges` API crash when a Syskeeper Forwarder bridge exists.
```
